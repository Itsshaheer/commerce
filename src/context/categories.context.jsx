import { useEffect, useState } from "react";
import { createContext } from "react";


import { addCollectionAndDocuments } from "../firebase";
import { getCategoriesAndDocuments } from "../firebase";



export const CategoriesContext= createContext({
    categoriesMap:[],
})

export const CategoriesProvider = ({children}) =>{
    const [categoriesMap,setCategoriesMap]= useState({})
    useEffect(()=>{
        const  getCategoriesMap= async ()=> {
            const categoryMap= await getCategoriesAndDocuments()
            
            setCategoriesMap(categoryMap)
        }
        getCategoriesMap()
       
    }, [])
    const value={categoriesMap}
    return (
       <CategoriesContext.Provider value={value}> {children} </CategoriesContext.Provider>
    )
}
