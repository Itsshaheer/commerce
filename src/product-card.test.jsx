import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import '@testing-library/jest-dom';
import { CartContext } from './cart.context';
import ProductCard from './product-card';

// Mock CartContext
const mockAddItemToCart = jest.fn();

const mockCartContextValue = {
  addItemToCart: mockAddItemToCart,
};

const mockProduct = {
  name: 'Sample Product',
  price: 20,
  imageUrl: 'sample-image.jpg',
};

test('adds item to cart on button click', () => {
  render(
    <CartContext.Provider value={mockCartContextValue}>
      <ProductCard product={mockProduct} />
    </CartContext.Provider>
  );

  // Find and click the Add to Cart button
  const button = screen.getByText('Add To Cart');
  fireEvent.click(button);

  // Ensure the addItemToCart function was called with the correct product
  expect(mockAddItemToCart).toHaveBeenCalledTimes(1);
  expect(mockAddItemToCart).toHaveBeenCalledWith(mockProduct);
});
