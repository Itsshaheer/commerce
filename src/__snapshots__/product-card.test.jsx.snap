// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="product-card-container"
  >
    <img
      src="https://example.com/product.jpg"
    />
    <div
      class="footer"
    >
      <span
        class="name"
      >
        Test Product 
      </span>
      <span
        class="price"
      >
         100
      </span>
    </div>
    <button
      class="button-container inverted"
    >
      Add To Cart 
    </button>
  </div>
</DocumentFragment>
`;
